reproducibility_project/src/engines/cassandra/project.py

```diff
@@ -17,7 +19,363 @@ def __init__(self):
+    else:
+        Nvap = 0
+
+    N = Nliq + Nvap
+
+    Nlist = [[Nliq]]
+
+    equil_length = 40000
+    prod_length = 120000
+
+    filled_boxes = construct_system(job.sp)
+
+    liqbox_filled = filled_boxes[0]
+    if ensemble == "GEMC-NVT":
+        vapbox_filled = filled_boxes[1]
+        Nlist.append([Nvap])
+
+    ffname = job.sp.forcefield_name
+    ff = load_ff(ffname)
+    structure = ff.apply(compound)
+
+    Tmelt = 1000.0 * u.K
+
+    T = job.sp.temperature * u.K
+    P = job.sp.pressure * u.kPa
+
+    species_list = [structure]
+    cutoff = job.sp.r_cut * u.nm
+
+    seedslist = [
+        [7860904, 98601355],
+        [955793508, 417823039],
+        [254420642, 2130720464],
+        [58120272, 1465850411],
+        [757616664, 1940492980],
+        [966844679, 1326087693],
+        [1992175335, 1840317929],
+        [650725500, 646331893],
+        [1204247127, 1521385831],
+        [2000946981, 1969870819],
+        [1488434295, 648017520],
+        [1128424221, 1140005446],
+        [58870203, 2133009902],
+        [2024564019, 2014788324],
+        [133927152, 2052536489],
+        [23375750, 1951798462],
+    ]
+
+    seeds = seedslist[job.sp.replica]
+
+    cbmc_n_ins = 12
+    cbmc_n_dihed = 50
+
+    prop_freq = 10
+    coord_freq = 10
+    if ff.combining_rule == "geometric":
+        comb_rule = "geometric"
+    else:
+        comb_rule = "lb"
+
+    proplist = [
+        "energy_total",
+        "volume",
+        "nmols",
+        "pressure",
+        "density",
+        "mass_density",
+        "energy_angle",
+        "energy_dihedral",
+        "energy_intravdw",
+        "energy_intraq",
+        "energy_inter",
+        "energy_intervdw",
+        "energy_lrc",
+        "energy_interq",
+        "energy_recip",
+        "energy_self",
+        "enthalpy",
+    ]
+
+    meltsystem_liq = mc.System([liqbox_filled], species_list, [[Nliq]])
+
+    p_volume = 0.01
+    p_translate = 0.0
+    p_rotate = 0.0
+    p_regrow = 0.0
+    p_swap = 0.0
+
+    if molecule == "methaneUA":
+        if ensemble == "NPT":
+            p_translate = 0.99
+        else:
+            p_translate = 0.7
+            p_swap = 0.29
+    elif molecule == "pentaneUA" and ensemble == "GEMC-NVT":
+        p_swap = 0.2
+        p_translate = 0.27
+        p_rotate = 0.26
+        p_regrow = 0.26
+    elif molecule == "waterSPCE":
+        p_translate = 0.49
+        p_rotate = 0.5
+    else:
+        p_translate = 0.33
+        p_rotate = 0.33
+        p_regrow = 0.33
+
+    nvtmoves = mc.MoveSet("nvt", species_list)
+    nvtmoves.prob_rotate = p_rotate
+    nvtmoves.prob_translate = p_translate
+    nvtmoves.prob_regrow = p_regrow
+
+    moveset = mc.MoveSet(cass_ensembles[ensemble], species_list)
+    moveset.prob_volume = p_volume
+    moveset.prob_translate = p_translate
+    moveset.prob_rotate = p_rotate
+    moveset.prob_regrow = p_regrow
+    moveset.cbmc_n_ins = cbmc_n_ins
+    moveset.cbmc_n_dihed = cbmc_n_dihed
+
+    with job:
```

Comment:
If you want, you can remove this context management statement and add the `@flow.with_job` decorator at the top with your other flow directives. This behaves the same way as the
```py3
with job:
    ...
``` 
context manager, but saves you a level of indentation

See [here](https://docs.signac.io/projects/flow/en/latest/api.html#flow-with-job)